### DISCUSSION TITLE
Time to ReBrand?

### DISCUSSION THREAD

**Discussion by chazhenry** [airsdk/Adobe-Runtime-Support#2588] [General]
I've been using the SDK for more than a year and have completely reanimated a deceased but valuable application. I'm building PC and Mac applications that are better in many ways than the Flash version. 

I'm writing to say that when someone asks what it's written in, I'm sheepish to say Air or Actionscript. I think its time to rebrand. Lose 'Adobe' all together. Lose 'Air'. Lose 'Actionscript'. Make it Harman Synapse, Harman Catalyst, Harman Anything. 

The SDK is strong and getting stronger. The cross platform capability competes well with the field. The game loop is a huge advantage, but what dev would start a new project with Adobe Air? Just my $.02, but it's time to rebrand.

**Comment by chazhenry** [comment]
Crickets. Hope I didn't step on toes here.

**Reply by ajwfrost** [reply]
:-) I'm not sure whether this would be a possibility .. we are running this still on behalf of Adobe, the software is licensed from them and they retain all the rights over it including the new work that we do. (The benefit of this is that if Harman ever decided to stop supporting it, Adobe would be able to pass it all over to someone else to pick up).

The one thing we had been exploring is trying to open it up to new languages. Using transpiler or LLVM type technologies so that someone could write an app using a different language, but still using the AIR APIs and running in the same AIR runtime. Which of course is an activity that anyone could work on :-)

**Comment by chazhenry** [comment]
Disingenuous (imho) for Adobe to quit on it yet retain control. The value is in the runtime and multiplatform ability. With just a couple of new language features its a new product and a new day. Think Javascript -> TypeScript. I'd put my effort there instead of whole new languages. ANEs could bridge the other gap. If Adobe truly wishes success they should admit that their name is a detriment. As for control, that's disappointing to hear. The parent who gets custody should be the one who cares. Harman has shown that they do.

**Reply by hydroper** [reply]
Yeah, TS is a successor language to JS, not an incremental one, so that could work. I guess Harman will try using Royale Compiler from Josh?

**Reply by chazhenry** [reply]
Yes, and sharing the runtime engine. Like Kotlin using the JVM. So other languages are an option. Much potential, but I'm guessing we're in a small and semi-private corner of the world. I'm running on PC/Mac and MS Surface. When I share with fellow devs, they can't believe it.

**Reply by ajwfrost** [reply]
When I mentioned "new languages" I was pretty much thinking of TypeScript .. although there may still be some challenges with it from what I can see. But the idea being that you write using TypeScript, and then a compiler turns that either into ActionScript Byte Code (and hence embedded into a SWF like normal), or would go the JIT route to compile straight to an Android/iOS/desktop application.

We were looking to adopt the Royale compiler, as that was based on an earlier iteration of the current AIR AS3 compiler, but with some stuff taken out. So we'd have to get a way to put that stuff back in, but with a different mechanism since that other stuff is private. But then yes it would be good to just have the one compiler to maintain!

We can't know what will happen in the future, so I see Adobe's position as a fairly sensible one. We do still have regular catch-ups and discussions on all this, they are pleased at its direction, and it may be that they could consider a branding change, but I think it's a good safety net to have the joint approach with them. I wouldn't really say they have "control" over the development itself - there are a few limitations perhaps, but we get to decide pretty much everything in terms of roadmap, features, platforms etc. But they do still own the trademarks, I'd definitely have to get their say-so before trying to rebrand it (and would also have to check internally within Harman).

I can raise it with them next week to see what their initial reaction is :-)

**Reply by teotigraphix** [reply]
Hah, this is like 2012 all over again. The Royale compiler... 😅

**Reply by raresn** [reply]
Only this time, Andrew knows he is doing! :) Imagine the exposure and people jumping in to code if they could do TypeScript instead of AS3! 👍

**Reply by shaucky** [reply]
Been a long time… So there's no more follow-up on this matter?

**Comment by 2jfw** [comment]
For attracting more people, why not additionally making the AIR **Linux** capabilities open to everyone @ajwfrost ?

**Comment by shaucky** [comment]
Fully support… AIR is a powerful and competitive framework. With IDEs like Animate and SDK, AIR has a strong development tool chain. But the Flash Player EOL brought Flash Platform a great blow… In this respect, Adobe did what worst.
 
I had wondered why AIR have low market occupancy, someone answered me, "adobe air is something very old…" And someone told me, "We just follow Jobs and what Jobs say what we believe." 
 
One of my classmate who focuses on JavaScript always query me why i still insist in ActionScript, neither of us could convince the other… Supporting other languages may be more beneficial.
 
I have find a description in the airsdk.dev referance:  A future version of the language is likely to have typed arrays. With some signes in github, it seems Harman has a plan to update ActionScript language version? i'm anticipating.
 
Rebrand is important for AIR.

**Comment by hydroper** [comment]
The `flash` package could possibly be aliased to a new package if the language supported something similiar to Rust's `use`. `use` is already a reserved word anyway.

For example:

```as3
package catalyst.utils {
    public use flash.utils.*;
}
```

This would export things like `ByteArray` and `setTimeout` efortlessly to `catalyst.utils`.

I think this feature can be implemented in the compiler, but when outputting ABC or SWF, the exports are erased, since ABC has no way of aliasing items in such way.

**Reply by shaucky** [reply]
Yeah, rebrand will encounter many historical baggage… AIR has a relatively low reputation. When i introduce AIR to others, if i want to set aside its relationship with Flash to avoid people's stereotypical impressions of Flash Player, it is still difficult to avoid mentioning things such as SWF files and `flash` class packages.

**Comment by linhdoha** [comment]
Changing the brand without the intrinsic nature of the technology being anything new or upgraded is quite useless. It's very easy to become a media crisis. I hope Harman will soon have creative and effective changes with this technology.

**Comment by Jason0247** [comment]
When I develop applications for Windows and iOS platforms, few people would think that it is based on AIR implementation, and even think that AIR development cannot produce such good things.
I personally think that branding is one aspect, but the other is the lack of a powerful compilation environment, such as tools like XCODE.
In fact, early FlashBuilder was very good, but I don't know why it was phased out later on.
I have also been looking forward to AIR's support for HUAWEI Harmony OS.

**Comment by GeneralVimes** [comment]
I like the eyes of a person who I'm talking to, when they ask "Cool game! What do you use to make it?" and I answer "It's Flash!" :D

Well, to be hones, I do this seldom, I tell that it's Harman AIR
